packages/ui/src/components/va-aspect-ratio/VaAspectRatio.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <div class="va-aspect-ratio">
+    <div v-if="stylesComputed" :style="stylesComputed" />
+    <slot />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed, type PropType } from 'vue'
+
+import { useComponentPresetProp } from '../../composables'
+
+export default defineComponent({
+  name: 'VaAspectRatio',
+
+  props: {
+    ...useComponentPresetProp,
+    ratio: {
+      type: [Number, String] as PropType<number | 'auto'>,
+      default: 'auto',
+      validator: (v: number | 'auto') => {
+        if (typeof v === 'number') {
+          return v > 0
+        }
+
+        return v === 'auto'
+      },
+    },
+    contentHeight: { type: Number, default: 1 },
```

Comment:
Not height/width?

Reply:
Yes, exactly content width-height. For the `aspect-ration` here will be separate `width` prop.